### PR TITLE
Test OTEL collector in xdock

### LINKED ISSUE #2155: Setup Crossdock for testing
We could deploy OTC (as agent and collector) in our CI as a separate crossdock job. The data flow would look like this otel-agent->otel-collector->jaeger-collector->storage.

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #2155

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

File: cmd/opentelemetry-collector/app/defaults/default_config.go [hunk 1]
```diff
@@ -90,6 +93,10 @@ func createReceivers(zipkinHostPort string, factories config.Factories) configmo
 			},
 		},
 	}
+	jaeger.RemoteSampling = &jaegerreceiver.RemoteSamplingConfig{
```
Comment (objectiser) [inline cmd/opentelemetry-collector/app/defaults/default_config.go:96]: Should the OTC, when used as a Jaeger collector replacement, be exposing the agent ports?

File: crossdock/otel-agent-config.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+receivers:
+  jaeger:
+    remote_sampling:
+      host_endpoint: localhost:5778
+      fetch_endpoint: jaeger-collector:14250
+    protocols:
+      grpc:
```
Comment (objectiser) [inline crossdock/otel-agent-config.yml:7]: Should grpc and thrift_http be here, if providing the role of the jaeger agent?

File: crossdock/docker-compose.yml [hunk 3]
```diff
@@ -82,6 +82,8 @@ services:
         image: jaegertracing/test-driver
         ports:
             - "8080"
+        environment:
+            - JAEGER_COLLECTOR_HC_HOST_PORT=jaeger-collector:${JAEGER_COLLECTOR_HC_PORT}
```
Comment (yurishkuro) [inline crossdock/docker-compose.yml:86]: test-driver is unchanged in this PR, what effect does giving it this new env var has?

File: crossdock/otel-agent-config.yml (hunk 2 above)
Comment (pavolloffay) [reply crossdock/otel-agent-config.yml:7]: I am not sure if we can disable `grpc` I think the sampling manager depends on it in the OTEL receiver. 

`thrift_http` can be disabled though.

File: crossdock/docker-compose.yml (hunk 3 above)
Comment (pavolloffay) [reply crossdock/docker-compose.yml:86]: This env var is consumed by `test_driver`. 

OTEL collector used a different health check port than Jaeger collector hence we have to specify it here.

File: cmd/opentelemetry-collector/app/defaults/default_config.go (hunk 1 above)
Comment (pavolloffay) [reply cmd/opentelemetry-collector/app/defaults/default_config.go]: Good point, I have disabled it for now.

File: crossdock/docker-compose.yml [hunk 4]
```diff
@@ -29,6 +29,8 @@ services:
         image: jaegertracing/xdock-go
         ports:
             - "8080-8082"
+        environment:
+            - AGENT_HOST_PORT=jaeger-agent:6831
```
Comment (pavolloffay) [inline crossdock/docker-compose.yml:33]: This going to be fixed in https://github.com/jaegertracing/jaeger-client-go/pull/500

**Comment by pavolloffay** [comment]
The build is green. Can anybody re-review?

**Pull request merged**